pkg/services/ngalert/notifier/encryption.go

```diff
@@ -0,0 +1,108 @@
+package notifier
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/store"
+	"github.com/grafana/grafana/pkg/services/secrets"
+)
+
+type Encryption interface {
+	LoadSecureSettings(ctx context.Context, orgId int64, receivers []*definitions.PostableApiReceiver) error
+	getDecryptedSecret(r *definitions.PostableGrafanaReceiver, key string) (string, error)
+	Encrypt(ctx context.Context, payload []byte, opt secrets.EncryptionOptions) ([]byte, error)
+}
+
+type encryptionImpl struct {
```

Comment:
```suggestion
type encryption struct {
```

with the above, you can remove the `impl` suffix.